src/browser/auth/Login.react.js

```diff
@@ -16,16 +16,14 @@ export default class Login extends Component {
     msg: PropTypes.object.isRequired
   }
 
-  onFormSubmit(e) {
+  async onFormSubmit(e) {
     e.preventDefault();
-    const {actions, auth: {form}} = this.props;
-    actions.login(form.fields)
-      .then(({error, payload: maybeValidationError}) => {
-        if (error)
-          focusInvalidField(this, maybeValidationError);
-        else
-          this.redirectAfterLogin();
-      });
+    const {actions: {login}, auth: {form}} = this.props;
+    const {error, payload: maybeValidationError} = await login(form.fields);
+    if (error)
+      focusInvalidField(this, maybeValidationError);
```

Comment:
The error is returned from the Promise which is created by the Redux promise middleware (the promise is never rejected).

I agree with the rest but that is IMO outside of the scope of this PR as this code and behavior is already in master.